infrastructure/containerd/image_service_integration_test.go

```diff
@@ -0,0 +1,215 @@
+		Use:          models.ImageUseVolume,
+	}
+	mounts, err = imageSvc.PullAndMount(ctx, fakePull)
+	Expect(err).To(HaveOccurred())
+	Expect(mounts).To(BeNil())
+
+	testImageMounted(ctx, imageSvc, testImageMountOptions{
+		ImageName: getTestVolumeImage(),
+		Owner:     inputGetAndMount.Owner,
+		Use:       models.ImageUseVolume,
+		Expected:  true,
+	})
+
+	testImageMounted(ctx, imageSvc, testImageMountOptions{
+		ImageName: "definitely-not-there",
+		Owner:     inputGetAndMount.Owner,
+		Use:       models.ImageUseVolume,
+		Expected:  false,
+	})
+
+	img, err := client.ImageService().List(namespaceCtx)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(img)).To(Equal(1))
+	Expect(img[0].Name).To(Equal(getTestVolumeImage()))
+
+	snapshotExists := false
+	err = client.SnapshotService(testSnapshotter).Walk(namespaceCtx, func(walkCtx context.Context, info snapshots.Info) error {
+		if info.Name == expectedSnapshotName {
+			snapshotExists = true
+		}
+
+		return nil
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(snapshotExists).To(BeTrue(), "expect snapshot with name %s to exist", expectedSnapshotName)
+
+	leases, err := client.LeasesService().List(namespaceCtx)
+	Expect(len(leases)).To(Equal(1))
+	Expect(leases[0].ID).To(Equal(expectedLeaseName), "expect lease with name %s to exists", expectedLeaseName)
+
+	inputGet.ImageName = testImageKernel
+
+	err = imageSvc.Pull(ctx, inputGet)
+	Expect(err).NotTo(HaveOccurred())
+
+	exists, err := imageSvc.Exists(ctx, &ports.ImageSpec{
+		ImageName: testImageVolume,
+		Owner:     testOwnerUsageID,
+	})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(exists).To(BeTrue())
+
+	mounts, err = imageSvc.PullAndMount(ctx, &ports.ImageMountSpec{
+		ImageName:    testImageKernel,
+		Owner:        testOwnerUsageID,
+		Use:          models.ImageUseKernel,
+		OwnerUsageID: testOwnerUsageID,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(mounts).NotTo(BeNil())
+	Expect(len(mounts)).To(Equal(1))
+
+	err = client.ImageService().Delete(namespaceCtx, testImageKernel)
+	Expect(err).NotTo(HaveOccurred())
+
+	err = client.ImageService().Delete(namespaceCtx, testImageVolume)
+	Expect(err).NotTo(HaveOccurred())
+
+	exists, err = imageSvc.Exists(ctx, &ports.ImageSpec{
+		ImageName: testImageKernel,
+		Owner:     testOwnerUsageID,
+	})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(exists).To(BeFalse())
+}
+
+func testCreateClient(t *testing.T) (*ctr.Client, context.Context) {
+	addr := os.Getenv("CTR_SOCK_PATH")
+	client, err := ctr.New(addr)
+	Expect(err).NotTo(HaveOccurred())
+
+	ctx := context.Background()
+
+	serving, err := client.IsServing(ctx)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(serving).To(BeTrue())
+
+	return client, ctx
+}
+
+func runContainerDTests() bool {
+	testCtr := os.Getenv("CTR_SOCK_PATH")
+	return testCtr != ""
+}
+
+func getTestVolumeImage() string {
+	envImage := os.Getenv("CTR_TEST_VOL_IMG")
+	if envImage != "" {
+		return envImage
+	}
+
+	return testImageVolume
+}
+
+func getTestKernelImage() string {
```

Comment:
this func isn't being used, I'm guessing you wanted it in place of all the spots where you have `testImageKernel`?